src/overlays/gamestates/ovl_daytelop/z_daytelop.c

```diff
@@ -1,22 +1,256 @@
+
+    temp_v0_2 = D_80815FF0;
+    new_var = this->fadeInState;
+    if (new_var == DAYTELOP_HOURSTEXT_FADEIN) {
+        s16 abs_temp = (s16)(ABS_ALT(this->alpha - 0xFF) / temp_v0_2);
+
+        this->alpha += abs_temp;
+        temp_v0_2--;
+        D_80815FF0 = temp_v0_2;
+
+        if (temp_v0_2 == 0) {
+            this->fadeInState = DAYTELOP_HOURSTEXT_ON;
+            this->alpha = 255;
+        }
+    }
+}
+#else
+void Daytelop_Update(DaytelopContext* this, GameState* gameState);
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_daytelop/Daytelop_Update.s")
+#endif
+
+void Daytelop_Draw(DaytelopContext* this) {
+    GraphicsContext* gfxCtx = this->common.gfxCtx;
+
+    OPEN_DISPS(gfxCtx);
+    func_8012C628(this->common.gfxCtx);
+
+    if (gSaveContext.day >= 9) {
+        // Draw a white screen
+        gDPSetRenderMode(POLY_OPA_DISP++, G_RM_XLU_SURF, G_RM_XLU_SURF2);
+        gDPSetCombineMode(POLY_OPA_DISP++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+        gDPFillRectangle(POLY_OPA_DISP++, 0, 0, SCREEN_WIDTH, SCREEN_HEIGHT);
+    }
+
+    func_8012C628(this->common.gfxCtx);
+
+    gDPSetRenderMode(POLY_OPA_DISP++, G_RM_OPA_SURF, G_RM_OPA_SURF2);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+
+    if (gSaveContext.day >= 9) {
+        gDPSetAlphaCompare(POLY_OPA_DISP++, G_AC_NONE);
+    }
+
+    // Draw the left side of the "Dawn of" texture
+    if (gSaveContext.day < 9) {
+        gDPLoadTextureBlock_4b(POLY_OPA_DISP++, sDayLeftTextures[CURRENT_DAY - 1], G_IM_FMT_I, 128, 64, 0,
+                               G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+    } else {
+        gDPLoadTextureBlock_4b(POLY_OPA_DISP++, sDayLeftTextures[ARRAY_COUNT(sDayLeftTextures) - 1], G_IM_FMT_I, 128,
+                               64, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+    }
+    gSPTextureRectangle(POLY_OPA_DISP++, 0x0080, 0x0134, 0x0280, 0x0234, G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
+
+    // Draw the right side of the "Dawn of" texture
+    if (gSaveContext.day < 9) {
+        gDPLoadTextureBlock_4b(POLY_OPA_DISP++, sDayRightTextures[CURRENT_DAY - 1], G_IM_FMT_I, 128, 64, 0,
+                               G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+    } else {
+        gDPLoadTextureBlock_4b(POLY_OPA_DISP++, sDayRightTextures[ARRAY_COUNT(sDayRightTextures) - 1], G_IM_FMT_I, 128,
+                               64, 0, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK,
+                               G_TX_NOLOD, G_TX_NOLOD);
+    }
+    gSPTextureRectangle(POLY_OPA_DISP++, 0x0280, 0x0134, 0x0480, 0x0234, G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
```

Comment:
```suggestion
    gSPTextureRectangle(POLY_OPA_DISP++, 160 << 2, 77 << 2, (160 + 128) << 2, (77 + 64) << 2, G_TX_RENDERTILE, 0, 0, 0x0400, 0x0400);
```
Same reason as above.